packages/@magic-sdk/provider/src/modules/base-module.ts

```diff
@@ -26,6 +26,18 @@ export class BaseModule {
    * Emits promisified requests to the Magic `<iframe>` context.
    */
   protected request<ResultType = any, Events extends EventsDefinition = void>(payload: Partial<JsonRpcRequestPayload>) {
+    if (
+      this.sdk.thirdPartyWallet.isConnected &&
+      payload.method !== MagicPayloadMethod.IntermediaryEvent &&
+      payload.method !== MagicPayloadMethod.NFTCheckout &&
+      payload.method !== MagicPayloadMethod.Login
```

Comment:
I think this implementation might encounter some forward compatible issue in the future. Is it possible to list the events that needs to be routed?